### PR TITLE
SQL: Better handling of number parsing exceptions

### LINKED ISSUE #33622: SQL: numeric overflows should be handled in a more friendly way
At the moment, something like `SELECT 123456789098765432101-2 AS x` will throw a convoluted error with a long stacktrace. Only a one line error message is preferred.

```
sql> select 123456789098765432101-2 AS x;
Server error [Server encountered an error [Overflow]. [java.lang.ArithmeticException: Overflow
        at java.base/java.math.BigDecimal.longValueExact(BigDecimal.java:3429)
        at org.elasticsearch.xpack.sql.parser.ExpressionBuilder.visitIntegerLiteral(ExpressionBuilder.java:482)
        at org.elasticsearch.xpack.sql.parser.ExpressionBuilder.visitIntegerLiteral(ExpressionBuilder.java:106)
        at org.elasticsearch.xpack.sql.parser.SqlBaseParser$IntegerLiteralContext.accept(SqlBaseParser.java:5685)
        at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:70)
        at org.elasticsearch.xpack.sql.parser.SqlBaseBaseVisitor.visitNumericLiteral(SqlBaseBaseVisitor.java:503)
        at org.elasticsearch.xpack.sql.parser.SqlBaseParser$NumericLiteralContext.accept(SqlBaseParser.java:4868)
        at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:70)
        at org.elasticsearch.xpack.sql.parser.SqlBaseBaseVisitor.visitConstantDefault(SqlBaseBaseVisitor.java:398)
        at org.elasticsearch.xpack.sql.parser.SqlBaseParser$ConstantDefaultContext.accept(SqlBaseParser.java:3957)
        at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:70)
........
```

### PR DESCRIPTION
**Pull request by costin**
Add proper exceptions in case the parsing of numbers (too large, invalid
format) fails.

Close #33622

### CODE REVIEW

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/parser/ExpressionTests.java [hunk 1]
```diff
@@ -64,4 +72,14 @@ public void testLiteralIntegerWithByteValue() throws Exception {
         assertEquals(Integer.valueOf(Byte.MAX_VALUE), l.value());
         assertEquals(DataType.INTEGER, l.dataType());
     }
+
+    public void testLiteralIntegerInvalid() throws Exception {
+        ParsingException ex = expectThrows(ParsingException.class, () -> parser.createExpression("123456789098765432101"));
+        assertEquals(ex.getErrorMessage(), "Cannot parse number [123456789098765432101]");
```
Comment (astefan) [inline x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/parser/ExpressionTests.java]: Shouldn't this be a "Number X is too large" situation?

Comment (costin) [reply x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/parser/ExpressionTests.java]: Fixed that - makes the parsing code a bit more ugly but the description is more accurate.

**Review by matriv — APPROVED** [review]
LGTM

**Review by astefan — APPROVED** [review]
LGTM

**Pull request merged**